api/utils/grpc/interceptors/mfa.go

```diff
@@ -44,8 +45,16 @@ func WithMFAUnaryInterceptor(mfaCeremony mfa.CeremonyFn) grpc.UnaryClientInterce
 			return trace.Wrap(err)
 		}
 
-		err := invoker(ctx, method, req, reply, cc, opts...)
-		if !errors.Is(trail.FromGRPC(err), &mfa.ErrAdminActionMFARequired) {
+		err = invoker(ctx, method, req, reply, cc, opts...)
+		if err == nil {
+			return nil
+		} else if mfaResp != nil && mfaResp.GetWebauthn() != nil {
+			// TODO(Joerger): DELETE IN v18.0.0
+			// If we passed an WebAuthn response and got an error, it may be that the
+			// backend WebAuthn session data got overwritten. Fallback to retrying with
```

Comment:
This is a strong assumption to make, we shouldn't be guessing. We are also assuming WebAuthn only ever fails this way, which isn't true.

Same for other places where this happens.